esp/logging/loggingagent/modularlogagent/README.md

```diff
@@ -0,0 +1,380 @@
+
+It is possible, if not likely, that one instance of this module will be expected to manage content in multiple files. The use of variable substitutions in the filepath makes it conceivable that every transaction could be written to a new file. To avoid unconstrained use of system resources, a limit on the number of files a module instance has open at one time is imposed through the configuration. It should only be necessary to change the default value when more than one file is expected. For example, multiple files would be expected if the *filepath* property includes a reference to the ESDL method name.
+
+To prevent output files from growing to a difficult to manage size, file rollover criteria can be given. Rollover on date change, a pattern seen elsewhere in the platform, is enabled by default. This may be insufficient for heavily used services, which is why rollover by file size may be enabled.
+
+## Usage
+
+The modular log agent supports two configuration scenarios.
+
+In the ***inline configuration*** scenario, an instance of **LogAgent** contains all available information for the agent. *Modular* and *agent integration* properties coexist in the same instance. Everything is in one place; the configuration is ***inline***.
+
+In the ***split configuration*** scenario, an instance of **LogAgent** contains the *agent integration* properties plus a reference to a separate file specifying *modular* properties. One configuration is derived from multiple sources; the configuration is ***split***.
+
+Logging configurations are presented to an ESP from multiple sources. Each configuration scenario is optimized for one or more of these sources, but may be applied to each source.
+
+### Legacy
+
+The ESP process may read an *esp.xml* configuration file to configure itself and its services. Traditionally this file is auto-generated from an *environment.xml* file created using *configmgr*. The ability for *configmgr* to configure constructs of which it is unaware is limited. Configurations specified with *configmgr* generally use a split configuration, but an inline configuration may be used to request default agent behavior.
+
+*Configmgr* includes a component named *modularlogagent*. This component is used to configure agents from any plugin using the modular agent framework. Reuse by multiple plugins requires little to no *modular* property knowledge exists in the component. The component supports the specification of *agent integration* properties, a split configuration property file, and (to support the default inline configuration option) a trace logging limit.
```

Comment:
This sentence doesn't quite parse right " Reuse by multiple plugins requires little to no *modular* property knowledge exists in the component."